cmd/root_test.go

```diff
@@ -124,14 +155,13 @@ func TestWarmupFailReadiness(t *testing.T) {
 	deleteFile("alive")
 	deleteFile("ready")
 
-	// we simulate a failure with no requests being sent by setting the port to a non-functional one
-	// we set the readiness port to the functional one (8080) so that health check passes
+	// we simulate a failure in the target by setting the readiness path to a non existent one so that
```

Comment:
I think the functionality we test here is different than the one tested before. Maybe related to the way we treat the fail readiness flag as part of this PR.
In this test we "tie" the flag to target readiness. It's obvious that this would fail in that case but don't we need to see if it fails when readiness succeeds but there's an error and warmup didn't run at all?

Reply:
i added a new test to cover that